src/blockdata/script.rs

```diff
@@ -313,12 +316,19 @@ impl Script {
 
     /// Generates P2WPKH-type of scriptPubkey
     pub fn new_v0_wpkh(pubkey_hash: &WPubkeyHash) -> Script {
-        Script::new_witness_program(WitnessVersion::V0, &pubkey_hash.to_vec())
+        Script::new_witness_program(WitnessVersion::V0, &pubkey_hash[..])
     }
 
     /// Generates P2WSH-type of scriptPubkey with a given hash of the redeem script
     pub fn new_v0_wsh(script_hash: &WScriptHash) -> Script {
-        Script::new_witness_program(WitnessVersion::V0, &script_hash.to_vec())
+        Script::new_witness_program(WitnessVersion::V0, &script_hash[..])
+    }
+
+    /// Generates P2TR-type of scriptPubkey from a provided public key value,
+    /// tweaking the key with its own TapTweak-tagged hash according to BIP-341
+    pub fn new_v1_p2tr<C: Verification>(secp: &Secp256k1<C>, internal_key: XOnlyPublicKey, merkle_root: Option<TapBranchHash>) -> Script {
```

Comment:
In e49a6d98f5b1bdf7bafe130777f9d8ce094d4760:

I think this name should be reserved for a conversion from a `TweakedKey` to a script, which is the cheapest and (maybe) most common case. Then users with other key types can either obtain one by tweaking (or `assume_dangerous` :P).

We probably also should have a method like this, which takes an xonly key and a merkle root, but it should be called something longer, `new_v1_p2tr_full` or something.

Reply:
Implemented

Reply:
>  which is the cheapest and (maybe) most common case.

I think the most common case would be p2tr(UntweakedPublicKey). I don't want to go into API design debates :P , but I think this saves the user from importing a trait `TapTweak` and learning about the trait API in the most common case. For descriptors, tr() descriptor has the internal untweaked key as the argument and the uncommon rawtr() descriptor would have the tweaked key.

I would prefer, `new_v1_p2tr(secp, untweaked_key, Option<MerkleRoot>)` and `new_v1_p2tr_raw(tweaked_key)`

Although this also has the benefit that the users know that the tweaking is being done internally in the API and they are expected to use the tweaked keypairs for signing. 

Not going to hold against this because we are offered the same functionality.